internal/jobqueue/dbjobqueue/dbjobqueue.go

```diff
@@ -208,6 +223,43 @@ func (q *dbJobQueue) Dequeue(ctx context.Context, jobTypes []string) (uuid.UUID,
 
 	return id, token, dependencies, jobType, args, nil
 }
+func (q *dbJobQueue) DequeueByID(ctx context.Context, id uuid.UUID) (uuid.UUID, []uuid.UUID, string, json.RawMessage, error) {
+	// Return early if the context is already canceled.
+	if err := ctx.Err(); err != nil {
+		return uuid.Nil, nil, "", nil, jobqueue.ErrDequeueTimeout
+	}
+
+	conn, err := q.pool.Acquire(ctx)
+	if err != nil {
+		return uuid.Nil, nil, "", nil, fmt.Errorf("error connecting to database: %v", err)
+	}
+
+	var jobType string
+	var args json.RawMessage
+	token := uuid.New()
+
+	err = conn.QueryRow(ctx, sqlDequeueByID, token, id).Scan(&token, &jobType, &args)
+	if err == pgx.ErrNoRows {
+		return uuid.Nil, nil, "", nil, fmt.Errorf("job '%v' isn't pending", id)
```

Comment:
err needs to be `jobqueue.ErrNotPending`

Reply:
Right, fixed, thanks!